announcer.rb

```diff
@@ -138,48 +146,95 @@ def play(file_data)
     file_path = "voice/#{dir}s/#{name}.wav"
     if File.file? file_path
       system AUDIO_COMMAND, file_path
+      record_log_entry(PRESENT_TEXT_FILE, name, dir)
     elsif file_data.to_a[1]
       _, route_id = file_data.to_a[1]
       file_path = "voice/#{dir}s/#{route_id}/#{name.tr '/', '-'}.wav"
       if File.file? file_path
         system AUDIO_COMMAND, file_path
-      else say(name, route_id)
+        record_log_entry(PRESENT_TEXT_FILE, name, "route #{route_id}")
+      else say(name, "route #{route_id}")
       end
-    else say(name, route_id)
+    else say(name, dir)
```

Comment:
This was a mistake. `route_id` wasn't defined here, so would've been `nil` (not a `NameError` because Ruby).